ee/lib/stripe/server.ts

```diff
@@ -0,0 +1,183 @@
+    currency: selectedEventType.currency,
+    payment_method_types: ["card"],
+    application_fee_amount: paymentFee,
+  };
+
+  const paymentIntent = await stripe.paymentIntents.create(params, { stripeAccount: stripe_user_id });
+
+  const payment = await prisma.payment.create({
+    data: {
+      type: PaymentType.STRIPE,
+      uid: uuidv4(),
+      bookingId: booking.id,
+      amount: selectedEventType.price,
+      fee: paymentFee,
+      currency: selectedEventType.currency,
+      success: false,
+      refunded: false,
+      data: Object.assign({}, paymentIntent, {
+        stripe_publishable_key,
+        stripeAccount: stripe_user_id,
+      }) as PaymentData as unknown as JsonValue,
+      externalId: paymentIntent.id,
+    },
+  });
+
+  const mail = new EventPaymentMail(
+    createPaymentLink(payment.uid, booking.user.name, booking.startTime.toISOString()),
+    evt,
+    booking.uid
+  );
+  await mail.sendEmail();
+
+  return payment;
+}
+
+export async function refund(
+  booking: {
+    id: number;
+    uid: string;
+    startTime: Date;
+    payment: {
+      id: number;
+      success: boolean;
+      refunded: boolean;
+      externalId: string;
+      data: JsonValue;
+      type: PaymentType;
+    }[];
+  },
+  calEvent: CalendarEvent
+) {
+  try {
+    const payment = booking.payment.find((e) => e.success && !e.refunded);
+    if (!payment) return;
+
+    if (payment.type != PaymentType.STRIPE) {
+      await handleRefundError({
+        event: calEvent,
+        booking: booking,
+        reason: "cannot refund non Stripe payment",
+        paymentId: "unknown",
+      });
+      return;
+    }
+
+    const refund = await stripe.refunds.create(
+      {
+        payment_intent: payment.externalId,
+      },
+      { stripeAccount: (payment.data as unknown as PaymentData)["stripeAccount"] }
+    );
+
+    if (!refund || refund.status === "failed") {
+      await handleRefundError({
+        event: calEvent,
+        booking: booking,
+        reason: refund?.failure_reason || "unknown",
+        paymentId: payment.externalId,
+      });
+      return;
+    }
+
+    await prisma.payment.update({
+      where: {
+        id: payment.id,
+      },
+      data: {
+        refunded: true,
+      },
+    });
+  } catch (e) {
+    console.log(e, "Refund failed");
+    await handleRefundError({
+      event: calEvent,
+      booking: booking,
+      reason: e.message || "unknown",
+      paymentId: "unknown",
+    });
+  }
+}
+
+async function handleRefundError(opts: {
+  event: CalendarEvent;
+  booking: { id: number; uid: string };
+  reason: string;
+  paymentId: string;
+}) {
+  console.log(`refund failed: ${opts.reason} for booking '${opts.booking.id}'`);
+  try {
+    await new EventOrganizerRefundFailedMail(
+      opts.event,
+      opts.booking.uid,
+      opts.reason,
+      opts.paymentId
+    ).sendEmail();
+  } catch (e) {
+    console.log("Error while sending refund error email", e);
```

Comment:
I would never catch and let something just trickle out as an error, this makes a false-positive.  At least `console.error` instead of normal logging